api/v1alpha1/linodemachine_types.go

```diff
@@ -28,13 +28,6 @@ import (
 
 // LinodeMachineSpec defines the desired state of LinodeMachine
 type LinodeMachineSpec struct {
-	// ProviderID is the unique identifier as specified by the cloud provider.
-	// +optional
-	ProviderID *string `json:"providerID,omitempty"`
```

Comment:
https://cluster-api.sigs.k8s.io/developer/providers/machine-infrastructure I think at least ProviderID at least needs to stay in the spec

Reply:
It was weird for me too, I also wanted to put `ProviderID` and `InstanceID` to status, but it seems spec is the official way :(